homeassistant/components/plugwise/climate.py

```diff
@@ -1,141 +1,152 @@
+    ):
         """Set up the Plugwise API."""
+        super().__init__(api, coordinator)
+
         self._api = api
+        self._name = name
+        self._dev_id = dev_id
+        self._loc_id = loc_id
+        self._model = model
         self._min_temp = min_temp
         self._max_temp = max_temp
-        self._name = name
-        self._direct_objects = None
-        self._domain_objects = None
-        self._outdoor_temperature = None
+
         self._selected_schema = None
         self._last_active_schema = None
         self._preset_mode = None
         self._presets = None
         self._presets_list = None
-        self._boiler_status = None
-        self._heating_status = None
-        self._cooling_status = None
-        self._dhw_status = None
+        self._boiler_state = None
+        self._heating_state = None
+        self._cooling_state = None
+        self._dhw_state = None
+        self._hvac_mode = None
         self._schema_names = None
         self._schema_status = None
-        self._current_temperature = None
-        self._thermostat_temperature = None
-        self._boiler_temperature = None
+        self._temperature = None
+        self._setpoint = None
         self._water_pressure = None
-        self._schedule_temperature = None
+        self._schedule_temp = None
         self._hvac_mode = None
+        self._single_thermostat = self._api.single_master_thermostat()
+        self._unique_id = f"{dev_id}-climate"
 
     @property
     def hvac_action(self):
-        """Return the current hvac action."""
-        if self._heating_status or self._boiler_status or self._dhw_status:
-            return CURRENT_HVAC_HEAT
-        if self._cooling_status:
-            return CURRENT_HVAC_COOL
-        return CURRENT_HVAC_IDLE
+        """Return the current action."""
+        if self._single_thermostat:
+            if self._heating_state or self._boiler_state:
+                return CURRENT_HVAC_HEAT
+            if self._cooling_state:
+                return CURRENT_HVAC_COOL
+            return CURRENT_HVAC_IDLE
+        if self._heating_state is not None or self._boiler_state is not None:
+            if self._setpoint > self._temperature:
+                return CURRENT_HVAC_HEAT
+            return CURRENT_HVAC_IDLE
 
     @property
     def name(self):
         """Return the name of the thermostat, if any."""
         return self._name
 
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+
+        device_information = {
+            "identifiers": {(DOMAIN, self._dev_id)},
+            "name": self._name,
+            "manufacturer": "Plugwise",
+            "model": self._model.replace("_", " ").title(),
+        }
+
+        if self._dev_id != self._api.gateway_id:
+            device_information["via_device"] = (DOMAIN, self._api.gateway_id)
+            del device_information["via_device"]
```

Comment:
Why do we set the key-value and then delete it directly after?

Reply:
Commit, mistake, already taken care of in the child-pr (I had exactly the same 'huh' moment yesterday)